.github/workflows/ci.yml

```diff
@@ -113,6 +113,8 @@ jobs:
           components: rust-src
       - name: Check doc-tests
         run: cargo xtask run-doc-test esp-hal ${{ matrix.device.soc }}
+      - name: Check documentation
+        run:  cargo xtask build-documentation --packages esp-hal --chips ${{ matrix.device.soc }}
```

Comment:
When we build the docs, are we checking that we have no warnings? Not checking warnings makes this step not that useful I think, as it's very unlikely that docs are left in an unbuildable state.

Reply:
> When we build the docs, are we checking that we have no warnings? 

Now we are! Good sugestion